src/main/java/org/broadinstitute/hellbender/transformers/IUPACReadTransformer.java

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.hellbender.transformers;
+
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.logging.OneShotLogger;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * A read transformer to convert IUPAC bases (i.e. non-ATCGs) to Ns
+ * Some references (like human hg38) contain IUPAC bases that can be propagated into the reads when decoding cram
+ * This transformation is done in-place without copying
+ */
+public class IUPACReadTransformer implements ReadTransformer {
+    private static final long serialVersionUID = 1L;
+    private boolean strictMode;
+    private OneShotLogger logger = new OneShotLogger(this.getClass());
+
+    public IUPACReadTransformer() {
+        this.strictMode = false;
+    }
+
+    public IUPACReadTransformer(final boolean strictMode) {
+        this.strictMode = strictMode;
+    }
+
+    @Override
+    public GATKRead apply(GATKRead read) {
+        final byte[] maybeTransformed = BaseUtils.convertIUPACtoN(read.getBases(), strictMode, false);
+        if (!Arrays.equals(read.getBases(), maybeTransformed)) {
+            logger.warn("At least one read contains IUPAC bases that have been transformed.  Read " + read.getName() + " contains: "
```

Comment:
This looks like an expensive string to construct for a few reasons. This should be pushed into a supplier (so () <- {STRING_STUFF} and you should add an overload in the one shot logger that takes a supplier and only feeds it to the underlying logger if it hasn't spoken yet:

```
    /**
     * Logs a message which is only to be constructed if the logging level is the {@link Level#WARN WARN} level.
     *
     * @param msgSupplier A function, which when called, produces the desired log message; the format depends on the
     *            message factory.
     * @since 2.4
     */
    void warn(Supplier<?> msgSupplier);
```